### PR TITLE
Modify collectAsState to collectAsStateWithLifecycle

### PR DESCRIPTION
**Pull request by pachuho**
Use collectAsStateWithLifecycle to save app resources when you don't need them, 
such as when they're in the background. 

The following tasks have been applied.

1. add dependency `androidx.lifecycle.lifecycle-runtime-compose`
2. modify flow to stateFlow in viewModel
3. modify collectAsState to collectAsStateWithLifecycle in Composable Screen

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt [hunk 1]
```diff
@@ -17,16 +17,26 @@
...
     gardenPlantingRepository: GardenPlantingRepository
 ) : ViewModel() {
-    val plantAndGardenPlantings: Flow<List<PlantAndGardenPlantings>> =
-        gardenPlantingRepository.getPlantedGardens()
+    val plantAndGardenPlantings: StateFlow<List<PlantAndGardenPlantings>> =
+        gardenPlantingRepository
+            .getPlantedGardens()
+            .stateIn(
+                viewModelScope,
+                SharingStarted.Lazily,
```
Comment (Jaehwa-Noh) [inline app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt:39]: You are collecting stateFlow with lifecycle,
https://github.com/android/sunflower/pull/976/files#diff-b218c68242277ca91d584e9db9dff7e2b1d5c32af496779fa559218ead47712eR68

but the SharingStarted.Lazily is never stop.
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-sharing-started/-companion/-lazily.html

Is this your intended behavior?

Comment (pachuho) [reply app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt]: @Jaehwa-Noh 
I corrected it, 
thank you again and if you have any other problems, please let me know.

File: app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt [hunk 2]
```diff
@@ -17,16 +17,25 @@
...
     gardenPlantingRepository: GardenPlantingRepository
 ) : ViewModel() {
-    val plantAndGardenPlantings: Flow<List<PlantAndGardenPlantings>> =
-        gardenPlantingRepository.getPlantedGardens()
+    val plantAndGardenPlantings: StateFlow<List<PlantAndGardenPlantings>> =
+        gardenPlantingRepository
+            .getPlantedGardens()
+            .stateIn(
+                viewModelScope,
+                SharingStarted.WhileSubscribed(),
```
Comment (bentrengrove) [inline app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt:38]: We generally recommend a timeout of 5 seconds on WhileSubscribed, this keeps it active during configuration changes.

https://medium.com/androiddevelopers/things-to-know-about-flows-sharein-and-statein-operators-20e6ccb2bc74

File: app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModel.kt [hunk 3]
```diff
@@ -42,6 +44,11 @@ class PlantDetailViewModel @Inject constructor(
     val plantId: String = savedStateHandle.get<String>(PLANT_ID_SAVED_STATE_KEY)!!
 
     val isPlanted = gardenPlantingRepository.isPlanted(plantId)
+        .stateIn(
+            viewModelScope,
+            SharingStarted.WhileSubscribed(),
```
Comment (bentrengrove) [inline app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModel.kt:49]: As above

**Review by bentrengrove — CHANGES_REQUESTED** [review]
Thanks for the PR! This looks really good, just one minor comment to fix

File: app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt (hunk 2 above)
Comment (pachuho) [reply app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GardenPlantingListViewModel.kt]: Thank you for a good idea!
I modified it.

**Review by bentrengrove — APPROVED** [review]
LGTM!

**Pull request merged**